internal/planmodifier/customplan.go

```diff
@@ -0,0 +1,49 @@
+package planmodifier
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func CustomPlanPlanModifier() planmodifier.List {
+	return &customPlanPlanModifier{}
+}
+
+type customPlanPlanModifier struct{}
+
+func (d *customPlanPlanModifier) Description(_ context.Context) string {
+	return "Ensures that custom_plan block is set if plan field's value is custom and vice versa."
+}
+
+func (d *customPlanPlanModifier) MarkdownDescription(ctx context.Context) string {
+	return d.Description(ctx)
+}
+
+func (d *customPlanPlanModifier) PlanModifyList(ctx context.Context, req planmodifier.ListRequest, resp *planmodifier.ListResponse) {
```

Comment:
Do we have tests for this? Could also add those later if needed.

Reply:
Added a unit test for now